src/Microsoft.DotNet.Interactive.Tests/Parsing/SubmissionParserTests.cs

```diff
@@ -366,6 +367,69 @@ public void Submissions_targeting_proxy_kernels_are_not_split_prior_to_sending()
                 .Be(code);
         }
 
+        [Fact]
+        public async Task proxyKernel_does_not_perform_split_if_all_parts_go_to_same_targetKernel_as_the_original_command()
+        {
+            
+            var commandAndEventReceiver = new BlockingCommandAndEventReceiver();
+            var commandAndEventSender = new RecordingKernelCommandAndEventSender();
+
+            commandAndEventSender.OnSend(commandOrEvent =>
+            {
+                if (commandOrEvent.Command is { } command)
+                {
+                    commandAndEventReceiver.Write(new CommandOrEvent( new CommandSucceeded(command)));
+                }
+            });
+            var proxyKernel = new ProxyKernel(
+                "proxyKernel",
+                commandAndEventReceiver,
+                commandAndEventSender);
+
+            var _ = proxyKernel.RunAsync();
+            var code = @"#i ""nuget:source1""
+#i ""nuget:source2""
+#r ""nuget:package1""
+#r ""nuget:package2""
+
+Console.WriteLine(1);";
+             await proxyKernel.SendAsync(new SubmitCode(code, proxyKernel.Name));
+
+            commandAndEventSender.Commands.Count.Should().Be(1);
```

Comment:
Do we want to assert that the command is in fact the original command?

Reply:
indeed!